src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -2425,6 +2425,69 @@ private static ObjectReference<Compilation> GetObservedCompilation(Solution solu
             return new ObjectReference<Compilation>(observed);
         }
 
+        [Fact]
+        [WorkItem(63834, "https://github.com/dotnet/roslyn/issues/63834")]
+        public void RecoverableTree_With()
+        {
+            using var workspace = CreateWorkspaceWithRecoverableSyntaxTreesAndWeakCompilations();
+
+            var pid = ProjectId.CreateNewId();
+            var did = DocumentId.CreateNewId(pid);
+
+            var sol = workspace.CurrentSolution
+                .AddProject(pid, "goo", "goo.dll", LanguageNames.CSharp)
```

Comment:
Can you switch this to a Theory to cover VB as well?

Reply:
Done for VB.